src/NuGet.Core/NuGet.Common/Utility/NuGetLogCodeUtility.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+
+namespace NuGet.Common
+{
+    public static class NuGetLogCodeUtility
+    {
+        public static bool AreVulnerabilitiesInCodes(IEnumerable<NuGetLogCode> nugetLogCodes)
```

Comment:
I don't feel like this API is a good design. It's trivial for callers to do a loop themselves, and a lot of .NET perf issues are specifically around loops and iterating collections. Specifically, avoiding `IEnumerable`, which causes memory allocations. If we remove this public API, then it's each caller's own responsibility for their own performance.

However, if we do keep this public API for some reason, from an API design point of view, perhaps a name that is more similar to LINQ methods will be easier for developers familiar with LINQ to guess/understand.

```suggestion
        public static bool AnyVulnerablePackage(IEnumerable<NuGetLogCode> nugetLogCodes)
```

Basically, I'm trying to think what's best for an API customer:

```csharp
NuGetLogCodeUtility.AnyVulnerablePackage(codes);
NuGetLogCodeUtility.HasVulnerablePackages(codes);
NuGetLogCodeUtility.ContainsVulnerablePackage(codes);
```

However, I'm not a fan of our utility class "pattern". I feel like it makes the call sites a bit unnatural. Would making this method an extension method make the call site better?

```csharp
IEnumerable<NuGetLogCode> codes = GetCodes();
bool showBar = codes.AnyVulnerablePackages();
showBar = codes.ContainsVulnerablePackages();
showBar = codes.AnyVulnerablePackageCode();
```

Reply:
Maybe let's drop the method and instead just store the vulnerable codes in a static?